README.md

```diff
@@ -1,20 +1,236 @@
+
+- `source` defines a data layer to be observed.
+- `destination` declares a function, which acts as a destination for data.
+- `operators` provide intermediate transformations of the data between the source and destination.
+
+A rule is expressed as JSON and multiple rules are included in a `_dlo_rules` list.
+
+```javascript
+window['_dlo_rules'] = [
+ {
+  id: 'fs-uservars-ceddl-user-all',
+  description: 'send all CEDDL user properties to FS.setUserVars',
+  source: 'digitalData.user.profile[0]',
+  operators: [ { name: 'flatten' } ],
+  destination: 'FS.setUserVars'
+},
+ {
+  id: 'fs-event-ceddl-transaction-id-total',
+  description: 'send CEDDL transaction transactionID and total properties to FS.event as an "Order Completed" event',
+  source: 'digitalData.transaction[(transactionID,total)]',
+  operators: [ { name: 'flatten' }, { name: 'insert', value: 'Order Completed' } ],
+  destination: 'FS.event'
+ }
+];
+```
+
+The above sample contains two rules that respectively send all properties in the `digitalData.user.profile` object to the function `FS.setUserVars` and sends specific properties from the `digitalData.transaction` object to the `FS.event` function.  The mechanics of how this is done is explained in the [Operator Tutorial](https://github.com/fullstorydev/fullstory-data-layer-observer/tree/main/docs/operator_tutorial.md) in greater detail.
+
+> **Tip**: While not required, a best practice is to include an `id` in every rule.  The `id` should uniquely identify the rule for troubleshooting purposes.  A `description` is also optional but often helps explain the intent of a rule.
+
+## Rule Options
+
+Each rule provides a set of options for configuration.  Options with an asterisk are required.
+
+| Option | Default | Description |
+| ------ | ------- | ----------- |
+| `source` * | `undefined` | Data layer source object using selector syntax. |
+| `destination` * | `undefined` | Destination function using selector syntax. |
+| `debug` | `false` | Set to true if the rule should print debug for each operator transformation. |
+| `description` | `undefined` | Text description of the rule. |
+| `id` | `undefined` | Unique identifier for the rule. |
+| `operators` | `[]` | List of operators that transform data before a destination. |
+| `readOnLoad` | `false` | Rule-specific override for `window[‘_dlo_readOnLoad’]`. |
+| `url` | `undefined` | Specifies a regular expression that enables the rule when the page URL matches. |
+
+> **Tip:** Use `url` to limit when data is read from a data layer and enhance performance.
+
+## Data Handling
+
+In version 1.0.x, data from a data layer is evaluated when DLO loads on the page.  The configuration option `window['_dlo_readOnLoad'] = true;` should be used for static data layers, which should exist on the page prior to loading DLO.  In the very near future, monitoring of dynamic changes in a data layer will be introduced.
+
+## Preview and Debugging Rules
```

Comment:
I think it might be a good idea to move this below the "source" and "destination" sections